### ISSUE TITLE
Support C# 10 in XamariniOS@2 task

### ISSUE THREAD

**Issue by benlings** [microsoft/azure-pipelines-tasks#16442]
## Required Information

*Type*: Feature

**Enter Task Name**: XamariniOS

## Environment
- Server - Azure Pipelines 
    
- Agent - Hosted
    
## Issue Description

The XamariniOS task uses the Mono version of MSBuild, which doesn't support C#10. The Hosted build agents in Azure Pipelines have Visual Studio 2022 for Mac installed. It is possible to use Mono to run the VS2022 version of MSBuild, which does support C#10. However, it doesn't seem like it's possible to configure the `XamariniOS` task to do this. 

I have tried the following:

- specify the following as an `input` parameter to the task:

```
    buildToolLocation: '/Library/Frameworks/Mono.framework/Versions/Current/Commands/mono "/Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/MSBuild.dll"'
```

This resulted in the error

```
Not found build tool: /Library/Frameworks/Mono.framework/Versions/Current/Commands/mono "/Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/MSBuild.dll"
```

- The workaround described here: https://medium.com/@tarmopr/how-to-fix-ci-build-with-azure-hosted-agent-for-xamarin-ios-project-that-uses-c-10-cc050ba7d459 . 

###

Desired functionality: Have some way of choosing the version of MSBuild included in Visual Studio Mac.

**Comment by vmapetr** [comment]
@benlings Thanks for reporting! We are working on more prioritized issues now but will get back to this one soon

**Comment by SaqibS** [comment]
Thanks! I've been wondering about this for months! Appreciate this issue just to understand the problem better, and the hacky work-around.

+1 on this feature request - We install latest Mono (6.12.0) using Boots, but it still gives the error about /langversion not supporting 10.0.

**Comment by OzaS** [comment]
> Any movement on this? the App compiles in VS for mac 2022 on the local mac. Going through the clean up efforts to use C# 10 only to be stumped but what seems to be something so basic.
> 
> Does anyone have a work around?
> 
> it seems to me if VS2022 for mac is on the image, we should be able to compile it another way.

The workaround in the first message works fine
But it's sad that the Task is not working for 4 month now

**Comment by wesleyscaldwell** [comment]
Yeah I bit the bullet and upgraded the project to Maui, which in some places was easier than expected. But I ran into several little bugs that made the upgrade extremely time consuming. 

I found that once you have the upgrade completed successfully and all your plugins and extensions upgraded correctly it works, but trying to diagnose what wasn't working took a lot longer than expected. The error messages were pretty misleading, I spent a lot of time trying to resolve symptoms and not the actual root issues.

I also had inconsistent issues with shell between platforms.
